tasks/actorstate/miner/extraction/state.go

```diff
@@ -0,0 +1,117 @@
+package extraction
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/chain/types"
+
+	"github.com/filecoin-project/lily/chain/actors/builtin/miner"
+	"github.com/filecoin-project/lily/tasks"
+	"github.com/filecoin-project/lily/tasks/actorstate"
+)
+
+type State interface {
```

Comment:
The state type used by miner extractors. Presents a (mockable) interface for gathering miner data for extraction